src/rust/cryptography-x509/src/extensions.rs

```diff
@@ -228,31 +233,84 @@ pub struct BasicConstraints {
     pub path_length: Option<u64>,
 }
 
+pub type SubjectAlternativeName<'a> = asn1::SequenceOf<'a, name::GeneralName<'a>>;
+pub type ExtendedKeyUsage<'a> = asn1::SequenceOf<'a, asn1::ObjectIdentifier>;
+
+pub struct KeyUsage<'a>(asn1::BitString<'a>);
```

Comment:
Similarly, there are certainly places in cryptography that should use this.

Reply:
I only found one deserialization site in Rust for this; included in https://github.com/pyca/cryptography/pull/9213/commits/f4a56648da2318db9f85564df05179bff0816ab7.